registration/include/pcl/registration/impl/ndt.hpp

```diff
@@ -200,11 +221,26 @@ NormalDistributionsTransform<PointSource, PointTarget>::computeDerivatives (Eige
       const Eigen::Matrix3d c_inv = cell->getInverseCov ();
 
       // Compute derivative of transform function w.r.t. transform vector, J_E and H_E in Equations 6.18 and 6.20 [Magnusson 2009]
-      computePointDerivatives (x);
+      computePointDerivatives (x, point_jacobian, point_hessian);
       // Update score, gradient and hessian, lines 19-21 in Algorithm 2, according to Equations 6.10, 6.12 and 6.13, respectively [Magnusson 2009]
-      score += updateDerivatives (score_gradient, hessian, x_trans, c_inv, compute_hessian);
+      score_pt += updateDerivatives (score_gradient_pt, hessian_pt, point_jacobian, point_hessian, x_trans, c_inv, compute_hessian);
     }
+
+    #ifdef _OPENMP
+    const int thread_idx = omp_get_thread_num();
+    #else
+    const int thread_idx = 0;
+    #endif
+
+    scores[thread_idx] += score_pt;
```

Comment:
Perhaps, we could use OMP's `reduction(+: score, score_gradient, hessian)`. They're supported since 2011's OpenMP 3.1

Reply:
I tried to use ```reduction```, but it seems it has issues with Eigen matrices and yields corrupted values. I'm not really sure, but I guess there are some alignment problems related to Eigen's optimization.